specs/interop/services.md

```diff
@@ -0,0 +1,83 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Superchain Backend](#superchain-backend)
+  - [Message Safety](#message-safety)
+- [Optimism Transaction Pool Policies](#optimism-transaction-pool-policies)
+  - [SuperchainMessagingPolicy](#superchainmessagingpolicy)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+## Superchain Backend
+
+The superchain backend provides the methods available for both
+the block builder and and verifier for message safety checks
+```go
+type MessageSafetyLabel int
+
+const (
+	Invalid MessageSafetyLabel = iota
+    Unsafe
+    CrossUnsafe
+	Safe
+	Finalized
+)
+
+type SuperchainBackend interface {
+    MessageSafety(id Identifier, payload []byte) (MessageSafetyLabel, error)
+}
+```
+
+The backend is instantiated per chain executing interop messages, maintaning a view
+against the peers in it's [dependency set](./dependency_set.md).
+
+### Message Safety
+
+This method MUST enforce the [Identifier](./messaging.md#message-identifier)'s chainId is present in the chain's [dependency set](./dependency_set.md).
+
+This method MUST enforce the attributes listed in the [Identifier](./messaging.md#message-identifier) matches the log & block attributes
+returned by peer's `eth_getLogs` request with the `blockNumber` and `logIndex` parameters.
+
+This method MUST enforce the provided payload matches the bytes computed when
+constructing the [MessagePayload](./messaging.md#message-payload) against the fetched log.
+
+This method MUST return the applicable [safety label](./verifier.md#safety) for the message, or `Invalid` 
+upon failures.
+
+This method MUST NOT enforce the [timestamp invariant](./messaging.md#timestamp-invariant) on the provided messages. The block builders
+and verifiers must locally do so based on their configured level of safety.
+
+## Optimism Transaction Pool Policies
+
+The `OptimismTxPoolPolicy` is an abstraction provided to the execution engine to apply additional
+logic to validation logic to the transaction pool.
+
+```go
+type OptimismTxPolicyStatus uint
+
+const (
+	OptimismTxPolicyInvalid OptimismTxPolicyStatus = iota
+	OptimismTxPolicyValid
+)
+
+type OptimismTxPoolPolicy interface {
```

Comment:
Why are there 2 different interfaces instead of just using a single one? I suppose the tradeoff is deserializing the calldata vs sending the whole tx. I don't think we want to send all txs thru this interface

Reply:
I'll remove this section as it's specific to op-geth but the txpool & block builder spec already clearly describes what the execution engine must do